src/plugins/data_explorer/public/utils/state_management/store.ts

```diff
@@ -62,6 +80,21 @@ export const getPreloadedStore = async (services: DataExplorerServices) => {
   // the store subscriber will automatically detect changes and call handleChange function
   const unsubscribe = store.subscribe(handleChange);
 
+  services.scopedHistory.listen(async (location, action) => {
+    const urlState = await loadReduxState(services);
+    const currentState = store.getState();
+
+    if (isEqual(urlState, currentState)) {
+      return;
+    }
+
+    // If the url state is different from the current state, then we need to update the store
+    // the state should have a view property if it was loaded from the url
+    if (action === 'POP' && urlState.metadata?.view) {
+      store.dispatch(hydrate(urlState as RootState));
+    }
```

Comment:
While the way this is coded has excellent readability, I wonder if we would get noticeable performance improvement if we moved the `isEqual` check inside the `action === 'POP' && urlState.metadata?.view` condition.

Reply:
That is a nice improvement :) Will make the change